astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -129,12 +129,10 @@ def cirs_to_icrs(cirs_coo, icrs_frame):
                                               distance=cirs_coo.distance,
                                               copy=False)
 
-        newxyz = intermedrep.to_cartesian().xyz
-        # roll xyz to last axis and add the barycentre position
-        newxyz = np.rollaxis(newxyz, 0, newxyz.ndim) + astrom['eb'] * u.au
-        # roll xyz back to the first axis
-        newxyz = np.rollaxis(newxyz, -1, 0)
-        newrep = CartesianRepresentation(newxyz).represent_as(SphericalRepresentation)
+        astrom_eb = CartesianRepresentation(
+            u.Quantity(np.rollaxis(astrom['eb'], -1, 0), u.au, copy=False),
+            copy=False)
```

Comment:
This construct is repeated often enough to make me think it's worth hiving off into a utility function.